lib/features/mailbox/presentation/mailbox_controller.dart

```diff
@@ -1336,6 +1353,60 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
     }
   }
 
+  void _allowSubaddressingMailboxAction(MailboxId mailboxId, Map<String, List<String>?>? currentRights) {
+    final accountId = mailboxDashBoardController.accountId.value;
+    final session = mailboxDashBoardController.sessionCurrent;
+    if (session != null && accountId != null) {
+      final allowSubaddressingRequest = SubaddressingMailboxRequest(
+          mailboxId,
+          currentRights,
+          MailboxSubaddressingState.enabled,
+          MailboxSubaddressingAction.allow
+      );
+
+      consumeState(_subaddressingMailboxInteractor.execute(session, accountId, allowSubaddressingRequest));
+    }
+  }
+
+  void _disallowSubaddressingMailboxAction(MailboxId mailboxId, Map<String, List<String>?>? currentRights) {
+    final accountId = mailboxDashBoardController.accountId.value;
+    final session = mailboxDashBoardController.sessionCurrent;
+    if (session != null && accountId != null) {
+      final disallowSubaddressingRequest = SubaddressingMailboxRequest(
+          mailboxId,
+          currentRights,
+          MailboxSubaddressingState.disabled,
+          MailboxSubaddressingAction.disallow
+      );
+
+      consumeState(_subaddressingMailboxInteractor.execute(session, accountId, disallowSubaddressingRequest));
+    }
+  }
```

Comment:
well, i copied it from `mailbox_controller.dart:_unsubscribeMailboxAction`, but i was wondering why both were necessary. should i keep only one (let's say `MailboxSubaddressingState` and try to completely remove the other ?

Reply:
These 2 methods perform an action, which need `SubaddressingMailboxRequest` as a parameter. Inside that request, consider if the action need to know current state of `MailboxSubaddressing, or/and the action need to be perform. If that action need both, why?

Reply:
the request carries information about the action to be performed, so the `subaddressingAction` parameter has to be kept.

`subaddressingState` was only used as a parameter in `SubaddressingSuccess` so that the toast notification could display the appropriate message. but `subaddressingAction` carries the same information as `subaddressingState` in the request, so i find it better to use `subaddressingAction` in `subaddressingSuccess` in order not to duplicate information.

doing so, `MailboxSubaddressingState` became completely unused, so i removed it.